py/runtime.c

```diff
@@ -1593,8 +1606,28 @@ mp_obj_t mp_import_from(mp_obj_t module, qstr name) {
 void mp_import_all(mp_obj_t module) {
     DEBUG_printf("import all %p\n", module);
 
-    // TODO: Support __all__
     mp_map_t *map = &mp_obj_module_get_globals(module)->map;
+
+    #if MICROPY_ENABLE_MODULE_ALL
+    mp_map_elem_t *elem = mp_map_lookup(map, MP_OBJ_NEW_QSTR(MP_QSTR___all__), MP_MAP_LOOKUP);
+    if (elem != NULL && mp_obj_is_type(elem->value, &mp_type_list)) {
+        // When __all__ is defined, we must explicitly load all specified
+        // symbols, possibly invoking the module __getattr__ function
+        size_t len;
+        mp_obj_t *items;
+        mp_obj_list_get(elem->value, &len, &items);
```

Comment:
```suggestion
        mp_obj_get_array(elem->value, &len, &items);
```

Could be a bit more flexible to allow tuple as well.

Reply:
You are absolutely right. This is actually CPython behaviour, and tuple have the big advantage of being treated as constants in bytecode and beeing fully frozen in ROM as `mp_rom_obj_tuple_t` for frozen modules, requiring no RAM storage.

I have force-pushed the new version with suggested changes.